### DISCUSSION TITLE
4-3 Python Programming with Rasterio - Unable to choose 16-core machine type

### DISCUSSION THREAD

**Discussion by ryangrammer** [ua-gist604b-f24/syllabus#11] [General]
Hello all,

When setting up my codespace for assignment 4-3, we are asked to change the machine type to "16-core". I am unable to select a 16-core machine. The only options I have are 2-core and 4-core. Is anyone else experiencing this issue? Could it be due to the fact that we are on free accounts? (https://freesewing.dev/tutorials/getting-started-codespaces#:~:text=As%20a%20free,hours%20per%20month) 

I tried using the 4-core machine but ran into the crash/recovery mode upon startup (as expected).

Thanks,
Ryan

**Comment by jenw-ua** [comment]
Yes, I'm having the same problem.  It did look like 16-core is available when you pay, so maybe something has changed?

**Comment by jenw-ua** [comment]
Hey Ryan,

Try it again and you should be able to select 8-core now. Professor Olsson made 8-core available to us. I was just able to complete the assignment. Hope it works for you!

**Reply by aaryno** [reply]
GitHub is not allowing us to use the larger machine types on an Educational promotion. To get through this and the rest of the assignments I had to connect this with my personal credit card.

**Reply by aaryno** [reply]
TLDR; use the 8-core

**Comment by travisz09** [comment]
I was also able to complete the assignment on my desktop by cloning the repository and using VSCode. I don't necessarily recommend this, though, because it took a lot of optimization of the conda environment and packages.
